src/components/RestockDialog.vue

```diff
@@ -0,0 +1,320 @@
+ */
+const productVariants = computed(() => {
+    if (getProductWithItsVariantsOnlyQuery.value !== null) {
+        return getProductWithItsVariantsOnlyQuery.value.product.variants.nodes
+    }
+
+    return []
+})
+
+/**
+ * Computed property for generating an array of amount options based on the maximum batch size.
+ */
+const amountOptions = computed(() => {
+    const defaultMaximumLength = 100
+    const length =
+        props.maximumBatchSize !== undefined
+            ? props.maximumBatchSize
+            : defaultMaximumLength
+
+    return Array.from({ length: length }, (_, index) => (index + 1).toString())
+})
+
+/**
+ * Helper function -- this is a rule -- to check if the input is not null.
+ *
+ * @param input - The input string to check.
+ * @returns True if the input is not null; an error message otherwise.
+ */
+function inputIsNotNull(input: string): boolean | string {
+    if (input !== null) {
+        return true
+    }
+
+    return 'The amount must be specified.'
+}
+
+/**
+ * Helper function -- this is a rule -- to check if the input is a natural number.
+ *
+ * @param input - The input string to check.
+ * @returns True if the input is a natural number; an error message otherwise.
+ */
+function inputIsNaturalNumber(input: string): boolean | string {
+    if (!inputIsNotNull(input)) {
+        return false
+    }
+
+    if (/^\d+$/.test(input) && parseInt(input) > 0) {
+        return true
+    }
+
+    return 'The input must be a natural number, e.g. 5.'
+}
+
+/**
+ * Helper function to check if the input is smaller than the maximum batch size.
+ *
+ * @param input - The input string to check.
+ * @returns True if the input is smaller than the maximum batch size; an error message otherwise.
+ */
+function inputIsSmallerThanMaximumBatchSize(input: string): boolean | string {
+    if (!inputIsNaturalNumber(input)) {
+        return false
+    }
+
+    if (props.maximumBatchSize === undefined) {
+        return true
+    }
+
+    const number = parseInt(input)
+
+    if (number <= props.maximumBatchSize && number > 0) {
+        return true
+    }
+
+    return (
+        'The amount may not exceed the specified maximum batch size of: ' +
+        props.maximumBatchSize
+    )
+}
+
+/**
+ * Ref for the preselected product variant ID.
+ */
+const selectedProductVariantId = ref<string | undefined>(
+    props.preselectedProductVariantId
+)
+
+/**
+ * Ref for the amount input.
+ */
+const amount = ref<string | null>(null)
+
+/**
+ * Computed property indicating whether there are changes in the component's state.
+ */
+const hasChanges = computed(() => {
+    return (
+        selectedProductVariantId.value !== props.preselectedProductVariantId ||
+        amount.value !== null
+    )
+})
+
+/**
+ * Resets the view model to its initial state.
+ * @returns Whether or not the view model was reset.
+ */
+function resetViewModel(): boolean {
+    if (hasChanges.value) {
```

Comment:
is there a reason for tracking the changes?
to my understanding, this just tracks if resetting is necessary, however, resetting if there are no changes should do no harm and would probably simplify the code as change tracking would no longer be required

Reply:
Yes, I have implemented tracking routinely. Now that I think about it, I like the suggestion of simplifying the code by taking that away.

Reply:
[The commit that "fixes" it...](https://github.com/MiSArch/frontend/pull/37/commits/433d32ec2bff6aef7e49908e2ccc6cced6d0ccda)